### PR TITLE
s/Makefile/CMakelists.txt

### PR DESCRIPTION
**Pull request by ianklatzco**
Mostly a straightforward port of the Makefile. I've tested lightly (running `grrshot` on a binary out of cb-multios, compiled for linux/decree, debug/release) and things appear to be correct.

### CODE REVIEW

**Review by pgoodman — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+# https://github.com/trailofbits/remill/blob/master/CMakeLists.txt
+project(grr)
+cmake_minimum_required (VERSION 3.2)
+
+enable_language(C)
+enable_language(CXX)
+enable_language(ASM)
+
+# What compilers are we using?
+set(CMAKE_C_COMPILER "clang-3.8")
```
Comment (pgoodman) [inline CMakeLists.txt]: This I think is the one change I'd like to see. Can you see if you can detect if the cmake compiler is not a variant of clang, and error out? Hard-coding clang 3.8 should not be necessary anymore.

Comment (ianklatzco) [reply CMakeLists.txt]: Done in 395631b.

File: CMakeLists.txt [hunk 2]
```diff
@@ -7,10 +7,13 @@ enable_language(C)
 enable_language(CXX)
 enable_language(ASM)
 
-# What compilers are we using?
-set(CMAKE_C_COMPILER "clang-3.8")
-set(CMAKE_CXX_COMPILER "clang++-3.8")
-set(CMAKE_ASM_COMPILER "clang-3.8")
+if (NOT ${CMAKE_C_COMPILER} MATCHES ".*clang")
```
Comment (pgoodman) [inline CMakeLists.txt]: I'd recommend instead trying this test: https://stackoverflow.com/questions/10046114/in-cmake-how-can-i-test-if-the-compiler-is-clang

**Review by pgoodman — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 2 above)
Comment (ianklatzco) [reply CMakeLists.txt]: Done in 7fb8fe9.

**Review by pgoodman — APPROVED** [review]

**Pull request merged**